shipyard-controller/handler/debughandler.go

```diff
@@ -0,0 +1,221 @@
+package handler
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/gin-gonic/gin"
+	apimodels "github.com/keptn/go-utils/pkg/api/models"
+	"github.com/keptn/keptn/shipyard-controller/common"
+	"github.com/keptn/keptn/shipyard-controller/models"
+)
+
+type IDebugHandler interface {
+	GetAllProjects(context *gin.Context)
+	GetSequenceByID(context *gin.Context)
+	GetAllSequencesForProject(context *gin.Context)
+	GetAllEvents(context *gin.Context)
+	GetEventByID(context *gin.Context)
+}
+
+type DebugHandler struct {
+	DebugManager IDebugManager
+}
+
+func NewDebugHandler(debugManager IDebugManager) *DebugHandler {
+	return &DebugHandler{
+		DebugManager: debugManager,
+	}
+}
+
+// GetAllProjects godoc
+// @Summary      Get all keptn projects
+// @Description  Get all keptn projects
+// @Tags         Project
+// @Success      200                  {object}  []apimodels.ExpandedProject     "ok"
+// @Failure      400                  {object}  models.Error                    "Bad Request"
+// @Failure      500                  {object}  models.Error                    "Internal error"
+// @Router       /debug/project [get]
+func (dh *DebugHandler) GetAllProjects(c *gin.Context) {
+	params := &models.GetProjectParams{}
+
+	projects, err := dh.DebugManager.GetAllProjects()
+
+	if err != nil {
+		SetInternalServerErrorResponse(c, fmt.Sprintf(InvalidRequestFormatMsg, err.Error()))
+		return
+	}
+
+	sort.Slice(projects, func(i, j int) bool {
+		return projects[i].ProjectName < projects[j].ProjectName
+	})
+
+	var payload = &apimodels.ExpandedProjects{
+		PageSize:    0,
+		NextPageKey: "0",
+		TotalCount:  0,
+		Projects:    []*apimodels.ExpandedProject{},
+	}
+
+	paginationInfo := common.Paginate(len(projects), params.PageSize, params.NextPageKey)
+	totalCount := len(projects)
+	if paginationInfo.NextPageKey < int64(totalCount) {
+		payload.Projects = append(payload.Projects, projects[paginationInfo.NextPageKey:paginationInfo.EndIndex]...)
+	}
+
+	payload.TotalCount = float64(totalCount)
+	payload.NextPageKey = paginationInfo.NewNextPageKey
+	c.JSON(http.StatusOK, payload)
+}
+
+// GetAllSequencesForProject godoc
+// @Summary      Get all sequences for specific project
+// @Description  Get all the sequences which are present in a project
+// @Tags         Sequence
+// @Param        project              path      string                    true "The name of the project"
+// @Success      200                  {object}  []models.SequenceState    "ok"
+// @Failure      400                  {object}  models.Error              "Bad Request"
+// @Failure      404                  {object}  models.Error              "not found"
+// @Failure      500                  {object}  models.Error              "Internal error"
+// @Router       /debug/project/{project} [get]
```

Comment:
is the path `debug/project/{project}` still up to date? If yes, the naming should be adapted to reflect that this is intended for retrieving sequences